sqlglot/parser.py

```diff
@@ -1329,6 +1329,27 @@ class Parser(metaclass=_Parser):
     # The style options for the DESCRIBE statement
     DESCRIBE_STYLES = {"ANALYZE", "EXTENDED", "FORMATTED", "HISTORY"}
 
+    # The style options for the ANALYZE statement
+    ANALYZE_STYLES = {
+        "VERBOSE": lambda self: "VERBOSE",
+        "FULL": lambda self: "FULL",
+        "SAMPLE": lambda self: "SAMPLE",
+        "SKIP_LOCKED": lambda self: "SKIP_LOCKED",
+        "LOCAL": lambda self: "LOCAL",
+        "NO_WRITE_TO_BINLOG": lambda self: "NO_WRITE_TO_BINLOG",
+        "BUFFER_USAGE_LIMIT": lambda self: f"BUFFER_USAGE_LIMIT {self._parse_number()}",
+    }
+
+    ANALYZE_EXPRESSION_PARSERS = {
+        "COMPUTE": lambda self: self._parse_analyze_statistics("COMPUTE"),
+        "ESTIMATE": lambda self: self._parse_analyze_statistics("ESTIMATE"),
+        "UPDATE": lambda self: self._parse_analyze_histogram("UPDATE"),
+        "DROP": lambda self: self._parse_analyze_histogram("DROP"),
+        "PREDICATE": lambda self: self._parse_analyze_columns("PREDICATE"),
+        "ALL": lambda self: self._parse_analyze_columns("ALL"),
+        "VALIDATE": lambda self: self._parse_command(),
```

Comment:
1. The arguments in this mapping look redundant. I'd consider removing them and using `self._prev.text.upper()` when entering the methods.
2. Why are we parsing a command for `VALIDATE`? If the parser does not support this clause's syntax, we should instead treat the whole input as a `Command`. Treating only a substring as a "command" is hacky.